internal/services/desktopvirtualization/virtual_desktop_host_pool_resource_test.go

```diff
@@ -160,7 +160,7 @@ resource "azurerm_virtual_desktop_host_pool" "test" {
   }
   lifecycle {
     ignore_changes = [
-      registration_info,
+      registration_info[0].expiration_date,
```

Comment:
we probably shouldn't have this here - i assume this is the case because the value is not returned from the API? instead of doing an ignore changes can we instead pull the value across from config by doing a d.Get() and using that value?

Reply:
how come there should be a perpetual diff for that property? it looks like a required property on the resource so shouldn't change unless the user changes it?

Reply:
The expiration_date attribute needs a future date/time to be valid and can't be set more than 30/31 days in the future.

The acceptance test resource provides this as:

`registration_info {
    expiration_date = timeadd(timestamp(), "48h")
  }`

As a result will always create a non-empty plan if we don't ignore changes in the test resource.

In practice this attribute will almost always be set to a dynamic future date as changing the expiration_date is required to generate a new valid token attribute (although this doesn't work as-is, but is fixed in https://github.com/hashicorp/terraform-provider-azurerm/pull/14134) so the test case is representative of real-world usage.

We could ignore non-empty plans, but that could mask other issues.

Reply:
ahh that makes sense, instead of generating the date in tf could we instead do it in go and then pass it into the test so we can remove the lifecycle ignore changes please?

Reply:
Done!